alphalens/tests/test_performance.py

```diff
@@ -444,3 +444,69 @@ def test_average_cumulative_return_by_quantile(self, before, after,
         expected = DataFrame(index=index, columns=range(-before,after+1), data=expected_vals)
         assert_frame_equal(avgrt, expected)
 
+    @parameterized.expand([(0, 2, False, 4,
+                            [[ 0.0,  -0.50, -0.75],
+                             [ 0.0,   0.0,    0.0],
+                             [ 0.0,   0.0,    0.0],
+                             [ 0.0,   0.0,    0.0],
+                             [ 0.0,   0.25,   0.5625],
+                             [ 0.0,   0.0,    0.0],
+                             [ 0.0,   0.50,   1.25],
+                             [ 0.0,   0.0,    0.0]]),
+                           (0, 3, True, 4,
+                            [[ 0.0, -0.5625, -1.015625, -1.488281],
+                             [ 0.0,  0.0,     0.0,       0.0],
+                             [ 0.0, -0.0625, -0.265625, -0.613281],
+                             [ 0.0,  0.0,     0.0,       0.0],
+                             [ 0.0,  0.1875,  0.296875,  0.339844],
+                             [ 0.0,  0.0,     0.0,       0.0],
+                             [ 0.0,  0.4375,  0.984375,  1.761719],
+                             [ 0.0,  0.0,     0.0,       0.0]]),
+                           (0, 3, False, 2,
+                            [[ 0.0, -0.25, -0.375,  -0.4375],
+                             [ 0.0,  0.0,  0.0,  0.0],
+                             [ 0.0,  0.375,  0.90625,   1.664062],
+                             [ 0.0,  0.0,  0.0,  0.0]]),
+                           (0, 3, True, 2,
+                            [[ 0.0, -0.3125, -0.640625, -1.050781],
+                             [ 0.0,  0.0,  0.0,  0.0],
+                             [ 0.0,  0.3125,  0.640625,  1.050781],
+                             [ 0.0,  0.0,  0.0,  0.0]]),
+                          ])
+    def test_average_cumulative_return_by_quantile_2(self, before, after,
+                                                   demeaned, quantiles,
+                                                   expected_vals):
+        dr = date_range(start='2015-1-15', end='2015-1-25')
+        dr.name = 'date'
+        tickers = ['A', 'B', 'C', 'D', 'E', 'F']
+        r1, r2, r3, r4 = (1.25, 1.50, 1.00, 0.50)
+        prices = DataFrame(index=dr, columns=tickers,
```

Comment:
Yeah, I'm not too sure myself. I would prefer the list comprehension because 1) less code duplication, 2) less wall-of-text.